homeworks/07-mars-viewer/src/store/store.ts

```diff
@@ -0,0 +1,28 @@
+import {configureStore, createSelector} from "@reduxjs/toolkit";
+import sols, {IImage, IImages} from "./sols";
+import {Selector} from "react-redux";
+import router, {Page} from "./router";
+
+const store = configureStore({
+    reducer: {
+		sols,
+		router
+    }
+});
+
+export type IState = ReturnType<typeof store.getState>
+
+// Selectors
+export const selectRoute: Selector<IState, Page> = state => state.router.route;
+export const selectSelectedSol: Selector<IState, number> = state => state.sols.selectedSol;
+export const selectIsLoading: Selector<IState, boolean> = state => state.sols.isLoading;
+export const selectImages: Selector<IState, IImages> = state => state.sols.images;
+export const selectFavorites: Selector<IState, number[]> = state => state.sols.favorite;
+export const selectImagesBySol = createSelector<IState, IImages, number, IImage[]>(
```

Comment:
passing types may be skipped here `createSelector` typings should be able to define without passing explicitly, but passing them is not a downside:)